### PR TITLE
[jsk_pcl_ros/line_segment_detector] Add consensus method of segmentation

### PR DESCRIPTION
**Pull request by iory**

### CODE REVIEW

**Comment by iory** [comment]
Please merge this.

**Comment by iory** [comment]
Please merge this.

**Comment by iory** [comment]
Please merge this.

**Comment by iory** [comment]
@k-okada
Please merge this.

**Comment by k-okada** [comment]
please compare with other methods, like hough or ransac...

--
◉ Kei Okada


2016-11-07 21:44 GMT+09:00 iory <notifications@github.com>:

**Comment by wkentaro** [comment]
@iory I think you should show the result with features you added with this PR.

**Comment by k-okada** [comment]
@iory ping

**Comment by YoheiKakiuchi** [comment]
ping @iory

**Comment by k-okada** [comment]
why you close this?

--
◉ Kei Okada


2017-03-30 16:31 GMT+09:00 iory <notifications@github.com>:

**Comment by iory** [comment]
In the range that I use, the method of RANSAC is enough

**Comment by k-okada** [comment]
even so, you'd better to complete PR, or I would answer "ok, let's wait few
more month, may be original commit will close these PR...." for
https://github.com/start-jsk/rtmros_tutorials/pull/506,
https://github.com/start-jsk/rtmros_tutorials/pull/474

--
◉ Kei Okada


2017-03-30 16:42 GMT+09:00 iory <notifications@github.com>:

**Comment by iory** [comment]
> you'd better to complete PR

This means reopen this PR?

**Comment by k-okada** [comment]
yes please

--
◉ Kei Okada


2017-03-30 20:48 GMT+09:00 iory <notifications@github.com>:

File: jsk_pcl_ros/cfg/LineSegmentDetector.cfg [hunk 1]
```diff
@@ -9,6 +9,9 @@ from math import pi
 
 gen = ParameterGenerator ()
 
+method_type = gen.enum([gen.const("RANSAC",   int_t,  0, "SAC_RANSAC"),
```
Comment (wkentaro) [inline jsk_pcl_ros/cfg/LineSegmentDetector.cfg]: `pcl::SAC_RANSAC`

File: jsk_pcl_ros/cfg/LineSegmentDetector.cfg [hunk 2]
```diff
@@ -9,6 +9,9 @@ from math import pi
...
+                        gen.const("LMEDS",   int_t,  1, "pcl::SAC_LMEDS")], "An enum for Segmentation Mehtods")
```
Comment (wkentaro) [inline jsk_pcl_ros/cfg/LineSegmentDetector.cfg]: Or `SAC_LMEDS`.

File: jsk_pcl_ros/cfg/LineSegmentDetector.cfg (hunk 1 above)
Comment (iory) [inline jsk_pcl_ros/cfg/LineSegmentDetector.cfg]: Keep consistency

File: jsk_pcl_ros/cfg/LineSegmentDetector.cfg (hunk 2 above)
Comment (iory) [inline jsk_pcl_ros/cfg/LineSegmentDetector.cfg]: Updated

File: jsk_pcl_ros/src/line_segment_detector_nodelet.cpp [hunk 3]
```diff
@@ -241,21 +245,55 @@ namespace jsk_pcl_ros
...
     rest_indices->indices = indices->indices;
     // use RANSAC to segment lines
-    pcl::SACSegmentation<PointT> seg;
-    seg.setOptimizeCoefficients (true);
-    seg.setModelType (pcl::SACMODEL_LINE);
-    seg.setMethodType (pcl::SAC_RANSAC);
-    seg.setDistanceThreshold (outlier_threshold_);
-    seg.setMaxIterations (max_iterations_);
-    seg.setInputCloud(cloud);
+    if(segmentation_update_) {
```
Comment (wkentaro) [inline jsk_pcl_ros/src/line_segment_detector_nodelet.cpp]: Why you can't do the operations in this block inside of configCallback?

File: jsk_pcl_ros/src/line_segment_detector_nodelet.cpp [hunk 4]
```diff
@@ -241,21 +245,55 @@ namespace jsk_pcl_ros
...
+      seg_.setOptimizeCoefficients (true);
+      seg_.setModelType (pcl::SACMODEL_LINE);
+      switch (segmentation_method_) {
+        case jsk_pcl_ros::LineSegmentDetector_SAC_RANSAC: {
+          seg_.setMethodType(pcl::SAC_RANSAC);
```
Comment (wkentaro) [inline jsk_pcl_ros/src/line_segment_detector_nodelet.cpp]: It seems that you can just set `seg_.setMethodType(segmentation_method_);`, seeing the documentation http://docs.pointclouds.org/trunk/method__types_8h_source.html. Is that possible?

**Comment by wkentaro** [comment]
@k-okada could you please merge this and release the next version of this package?

**Comment by k-okada** [comment]
done https://github.com/ros/rosdistro/pull/15253

--
◉ Kei Okada


2017-06-16 1:27 GMT+09:00 Kentaro Wada <notifications@github.com>:

**Pull request merged**